src/main/kotlin/com/avail/AvailRuntime.kt

```diff
@@ -0,0 +1,1814 @@
+						.trimIndent())
+			}
+			finally
+			{
+				levelOneSafeLock.lock()
+				try
+				{
+					incompleteLevelOneUnsafeTasks--
+					if (incompleteLevelOneUnsafeTasks == 0)
+					{
+						assert(incompleteLevelOneSafeTasks == 0)
+						incompleteLevelOneSafeTasks = levelOneSafeTasks.size
+						levelOneSafeTasks.forEach(Consumer { this.execute(it) })
+						levelOneSafeTasks.clear()
+					}
+				}
+				finally
+				{
+					levelOneSafeLock.unlock()
+				}
+			}
+		}
+		levelOneSafeLock.lock()
+		try
+		{
+			// Hasten the execution of pending Level One-safe tasks by
+			// postponing this task if there are any Level One-safe tasks
+			// waiting to run.
+			if (incompleteLevelOneSafeTasks == 0
+			    && levelOneSafeTasks.isEmpty())
+			{
+				assert(!levelOneSafetyRequested)
+				incompleteLevelOneUnsafeTasks++
+				execute(wrapped)
+			}
+			else
+			{
+				levelOneUnsafeTasks.add(wrapped)
+			}
+		}
+		finally
+		{
+			levelOneSafeLock.unlock()
+		}
+	}
+
+	/**
+	 * Request that the specified action be executed as a Level One-safe task at
+	 * such a time as there are no Level One-unsafe tasks running.
+	 *
+	 * @param priority
+	 *   The priority of the [AvailTask] to queue.  It must be in the range
+	 *   [0..255].
+	 * @param safeAction
+	 *   The action to execute when Level One safety is ensured.
+	 */
+	fun whenLevelOneSafeDo(priority: Int, safeAction: () -> Unit)
+	{
+		val task = AvailTask(priority)
+		{
+			try
+			{
+				safeAction.invoke()
+			}
+			catch (e: Exception)
+			{
+				System.err.println(
+					"\n\tException in level-one-safe task:\n\t${trace(e)}"
+						.trimIndent())
+			}
+			finally
+			{
+				levelOneSafeLock.lock()
+				try
+				{
+					incompleteLevelOneSafeTasks--
+					if (incompleteLevelOneSafeTasks == 0)
+					{
+						assert(incompleteLevelOneUnsafeTasks == 0)
+						levelOneSafetyRequested = false
+						incompleteLevelOneUnsafeTasks = levelOneUnsafeTasks.size
+						for (t in levelOneUnsafeTasks)
+						{
+							execute(t)
+						}
+						levelOneUnsafeTasks.clear()
+					}
+				}
+				finally
+				{
+					levelOneSafeLock.unlock()
+				}
+			}
+		}
+		levelOneSafeLock.lock()
+		try
+		{
+			levelOneSafetyRequested = true
+			if (incompleteLevelOneUnsafeTasks == 0)
+			{
+				incompleteLevelOneSafeTasks++
+				execute(task)
+			}
+			else
+			{
+				levelOneSafeTasks.add(task)
+			}
+		}
+		finally
+		{
+			levelOneSafeLock.unlock()
+		}
+	}
+
+	/**
+	 * Destroy all data structures used by this `AvailRuntime`.  Also
+	 * disassociate it from the current [Thread]'s local storage.
+	 */
+	fun destroy()
+	{
+		timer.cancel()
+		executor.shutdownNow()
+		ioSystem.destroy()
+		callbackSystem.destroy()
+		try
+		{
+			executor.awaitTermination(10, TimeUnit.SECONDS)
+		}
+		catch (e: InterruptedException)
+		{
+			// Ignore.
+		}
+		moduleNameResolver.clearCache()
+		moduleNameResolver.destroy()
+		modules = NilDescriptor.nil
+	}
+
+	// Schedule a fixed-rate timer task to increment the runtime clock.
+	init
```

Comment:
This can be removed if the timer val is initialized during declaration thusly:
```
	val timer = fixedRateTimer("timer for Avail runtime", true, period = 10) {
		clock.increment()
	}
```

Reply:
not sure if you are advocating creating a function `fixedRateTimer` here whose body is similar to the init block...